terraform/node_output.go

```diff
@@ -38,7 +39,12 @@ func (n *NodeApplyableOutput) References() []string {
 	var result []string
 	result = append(result, ReferencesFromConfig(n.Config.RawConfig)...)
 	for _, v := range result {
-		result = append(result, v+".destroy")
+		split := strings.Split(v, "/")
+		for i, s := range split {
+			split[i] = s + ".destroy"
+		}
+
+		result = append(result, strings.Join(split, "/"))
```

Comment:
Yes, definitely, I'm glad you caught this.

So, this is a lamentable example of where we're using strings when we should probably be using some sort of structured data. I actually want to look into that refactor shortly after this, but didn't want to merge the two since I think that'll end up being fairly significant.

In this case, the syntax `A/B` means "I depend on A, but if A is not available, I depend on B. If A is available, don't connect to B."

The use case for this is in expanded resources. There are three cases for a resource reference:

  1. `aws_instance.foo.bar` - Equivalent to `aws_instance.foo.0.bar`. See next.

  2. `aws_instance.foo.0.bar`, `aws_instance.foo.5.bar`, etc. (exact index) - In this case, you want to depend EXACTLY on a specific instance if it exists. However, if it doesn't exist (computed count), you want to depend on the thing that will expand into an exact instance. So, in this case, we set the depends on to be `aws_instance.foo.0/aws_instance.foo.N` where "N" is a special syntax used to denote "I will expand to more things at runtime."

  3. `aws_instance.foo.*.bar` (splat) - Straightforward, you depend on all `aws_instance.foo` instances. This creates a depends on of `aws_instance.foo.*` which every matching resource will advertise as something they can be depended on as.

The key thing is (2) above. The existing reference system didn't support this. **THE WAY IT WORKED BEFORE:**

  1. Everything would depend purely on the `aws_instance.foo` granularity (not count-specific)

  2. During count expansion (`DynamicExpand` in `transform_resource.go`, the old graph), we'd override `DependsOn` and `DependableName` to append the index if we have one. Then we'd reconnect.

This doesn't work in the new graph because some specific indexes (like orphans) are created outside the graph. I also wanted to lay the groundwork so that we can avoid cycles in computed counts if we know a specific index exists (in the state, for example). This isn't fixed yet with this new graph but we're heading in that direction and we'll have to anyways. 

In conclusion, the right long term solution is a structure rather than a string. I patched on the `/` syntax since a resource name can't contain a `/` so its a safe character to use. However, I do want to take a look at this again to clean it up, but wanted to save that for another time since that refactor will touch a lot more than just plan-time stuff.